pkg/scaling/resolver/scale_resolvers_test.go

```diff
@@ -240,6 +240,56 @@ func TestResolveNonExistingConfigMapsOrSecretsEnv(t *testing.T) {
 	}
 }
 
+func TestResolveAuthRefAndPodIdentity(t *testing.T) {
+	if err := corev1.AddToScheme(scheme.Scheme); err != nil {
+		t.Errorf("Expected Error because: %v", err)
+	}
+	if err := kedav1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		t.Errorf("Expected Error because: %v", err)
+	}
+	tests := []struct {
+		name                string
+		existing            []runtime.Object
+		soar                *kedav1alpha1.AuthenticationRef
+		podTemplateSpec     *corev1.PodTemplateSpec
+		expected            map[string]string
+		expectedPodIdentity kedav1alpha1.AuthPodIdentity
+		expectedError       error
+	}{
+		// ensure podIdentity is returned
+		{
+			name:                "foo",
+			expected:            make(map[string]string),
+			expectedPodIdentity: kedav1alpha1.AuthPodIdentity{Provider: kedav1alpha1.PodIdentityProviderNone},
+		},
+	}
```

Comment:
Do we need table tests for a single case?

Reply:
good point!

I have prematurely asked for a review, I am still looking if I should add more tests on top of https://github.com/kedacore/keda/pull/5255, my gut feeling is yes but if I see no reason then I will remove the table test structure.

Reply:
Nice! Let me know once it's ready to merge xD

Reply:
I came to the conclusion that `TestResolveAuthRefAndPodIdentity` was similar in execution to `TestResolveAuthRef` and instead enhanced by two testcases the already existing table test function.

https://github.com/kedacore/keda/blob/b4d862c248dc50bebd47d5d23dc4e415e2621b4b/pkg/scaling/resolver/scale_resolvers_test.go#L574-L611

ptal @JorTurFer, I think it's ready for a review now